junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/TempDirectory.java

```diff
@@ -80,12 +80,19 @@ public void beforeAll(ExtensionContext context) throws Exception {
 	 * with {@link TempDir @TempDir}.
 	 */
 	@Override
-	public void beforeEach(ExtensionContext context) throws Exception {
-		injectFields(context, context.getRequiredTestInstance(), ReflectionUtils::isNotStatic);
+	public void beforeEach(ExtensionContext context) throws RuntimeException {
```

Comment:
```suggestion
	public void beforeEach(ExtensionContext context) {
```

There is no need to switch to a `RuntimeException`. In fact, you can simply remove the `throws` declaration.